specification/paths/Returns-v1-returns-return_id-payment-checkout-url-payment_id.json

```diff
@@ -0,0 +1,52 @@
+{
+  "parameters": [
+    {
+      "$ref": "#/components/parameters/path-return_id"
+    },
+    {
+      "$ref": "#/components/parameters/path-payment_id"
+    },
+    {
+      "$ref": "#/components/parameters/query-hmac"
+    }
+  ],
+  "get": {
+    "tags": [
+      "Returns"
+    ],
+    "security": [
+      {
+        "OAuth2": [
+          "returns.manage"
+        ]
+      }
+    ],
+    "summary": "Get a return payment checkout url",
+    "description": "This endpoint requests a return payment checkout url to a payment provider",
+    "responses": {
+      "302": {
+        "description": "The return payment checkout url is returned in the Location header.",
+        "headers": {
+          "Location": {
+            "schema": {
+              "type": "string"
+            },
+            "description": "The return payment checkout url"
+          }
+        }
+      },
+      "403": {
+        "description": "The payment provider does not allow the return payment link to be created or HMAC signature is invalid."
+      },
+      "422": {
+        "description": "The payment link expired."
+      },
+      "404": {
+        "description": "The payment was not found."
+      },
```

Comment:
Just looking at the Controller action `ReturnPaymentController::checkoutUrl` (so not tracking down possible exceptions in different files), we currently throw a 422 exception when:
1. the Shop is not connected to Mollie
2. the ReturnPayment is not found
3. the ReturnPayment is already paid

If the payment link has expired, we just create a new one (ReturnPaymentController line 92-96). 

I think it makes more sense to throw a ModelNotFoundException when the ReturnPayment is not found in the API, so this part of the spec would stay valid (after the change in the API)

Reply:
That is correct! I will change the description

Reply:
@yoerriwalstra specified the description for case 1 and 3. 2 is 404